server/controllers/assess/statusUpdateController.ts

```diff
@@ -39,10 +40,22 @@ export default class StatusUpdateController {
       const { newStatus } = req.body
 
       try {
+        const statuses = await this.submittedApplicationService.getApplicationStatuses(req.user.token)
+
+        const hasStatusDetails = statuses.some(
+          status => Boolean(status.statusDetails?.length) && status.name === newStatus,
+        )
+
+        if (hasStatusDetails) {
+          return res.redirect(
+            paths.statusUpdateDetails.new({ id: applicationId, statusName: camelToKebabCase(newStatus) }),
+          )
+        }
```

Comment:
hmmmmm, how does this work? Are we only ever expecting to get one status back from getApplicationStatuses, but in an array? What happens if we get multiple back?

Reply:
`getApplicationStatuses` is a bit misleading as it's actually returning the statuses from the reference data and not the statuses associated with a particular application.

So we call `getApplicationStatuses` which will return us the equivalent of what's in `application-statuses.json` then we compare whether the selected status has 1 or more status details in the reference data and render a new page.

Statuses without status details will include an empty array and will therefore return false due to this check: `Boolean(status.statusDetails?.length)`